src/Workspaces/Core/Portable/Shared/TestHooks/YieldAwaitableExtensions.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Threading;
+
+namespace Microsoft.CodeAnalysis.Shared.TestHooks
+{
+    internal static class YieldAwaitableExtensions
```

Comment:
Would appreciate some comments describing what this class is used for in test code. Unless you encounter its use case in `AsynchronousOperationListenerProvider` I don't think its immediately obvious

Reply:
➡️ File was removed from this pull request since it already lives in the compiler layer